include/deal.II/sundials/n_vector.templates.h

```diff
@@ -253,7 +253,7 @@ namespace SUNDIALS
       template <
         typename VectorType,
         typename std::enable_if_t<is_serial_vector<VectorType>::value, int> = 0>
-      MPI_Comm get_communicator(N_Vector);
```

Comment:
Hmm... Stylistically I would try to avoid having overloads with different return types here.
What about overloading `get_communicator_as_void_ptr` for serial vectors instead? That one always return `void*` and returning `nullptr` is no problem then.